cmd/kubeapps-apis/plugins/resources/v1alpha1/common/plugin_test.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2022 the Kubeapps contributors.
+// SPDX-License-Identifier: Apache-2.0
+package common
+
+import (
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/vmware-tanzu/kubeapps/cmd/kubeapps-apis/plugins/pkg/pkgutils"
+	log "k8s.io/klog/v2"
+	"os"
+	"runtime"
+	"sigs.k8s.io/yaml"
+	"strings"
+	"testing"
+)
+
+func TestParsePluginConfig(t *testing.T) {
+	testCases := []struct {
+		name           string
+		pluginYAMLConf []byte
+		expectedConfig *ResourcesPluginConfig
+		expectedError  string
+	}{
+		{
+			name:           "non existing plugin-config file",
+			pluginYAMLConf: nil,
+			expectedConfig: &ResourcesPluginConfig{},
+			expectedError:  "",
+		},
+		{
+			name: "invalid plugin config",
+			pluginYAMLConf: []byte(`
+resources:
+  packages:
+    v1alpha1:
+      trustedNamespaces:
+        headerName: true
+      `),
+			expectedConfig: nil,
+			expectedError:  "json: cannot unmarshal",
+		},
+		{
+			name: "non-default, valid plugin config",
+			pluginYAMLConf: []byte(`
+resources:
+  packages:
+    v1alpha1:
+      trustedNamespaces:
+        headerName: "X-Consumer-Groups"
+        headerPattern: "^namespace:([\\w-]+)$"
+      `),
+			expectedConfig: &ResourcesPluginConfig{
+				TrustedNamespaces: TrustedNamespaces{
+					HeaderName:    "X-Consumer-Groups",
+					HeaderPattern: "^namespace:([\\w-]+)$",
+				},
+			},
+			expectedError: "",
+		},
+	}
+	opts := cmpopts.IgnoreUnexported(pkgutils.VersionsInSummary{})
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// TODO(agamez): env vars and file paths should be handled properly for Windows operating system
+			if runtime.GOOS == "windows" {
+				t.Skip("Skipping in a Windows OS")
+			}
+			filename := ""
+			if tc.pluginYAMLConf != nil {
+				pluginJSONConf, err := yaml.YAMLToJSON(tc.pluginYAMLConf)
+				if err != nil {
+					log.Fatalf("%s", err)
+				}
```

Comment:
Haven't you considered the use of `github.com/stretchr/testify` library (which I've checked is already included in the project) to ease the writing and reading of tests? This way you could replace these three lines with just a oneliner like `require.NotNil(t, err)`

Reply:
Definitely, I'll try to remember next time. In many cases, this block of code is just copy-pasted.
Thanks!